### PR TITLE
Enforce a complete profile to send request, message, or create new event

### PR DESCRIPTION
**Pull request by aapeliv**
Closes frontend part of #4387

Screenshot from 2024-09-08 15-05-52

I believe I have committed many frontend sins here... so I think it needs a bit more work.

**Web frontend checklist**
- [ ] Formatted my code with `yarn format`
- [ ] There are no warnings from `yarn lint --fix`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by bakeiro** [comment]
cool features!

File: app/web/components/ProfileIncompleteDialog/ProfileIncompleteDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                variant="body1"
+                href="https://help.couchers.org/hc/couchersorg-help-center/articles/1725919152-why-do-i-need-to-complete-my-profile-to-use-some-features"
+              >
+                please refer to this help page
+              </StyledLink>
+              . Thank you for your help!
+            </Trans>
+          </Typography>
+        </DialogContentText>
+      </DialogContent>
```
Comment (nabramow) [inline app/web/components/ProfileIncompleteDialog/ProfileIncompleteDialog.tsx]: This bit is causing a console error:

`
ProfileIncompleteDialog.tsx:23 Warning: validateDOMNesting(...): <p> cannot appear as a descendant of <p>.
    at p
    at Typography (webpack-internal:///./node_modules/@material-ui/core/esm/Typography/Typography.js:166:28)
    at WithStyles (webpack-internal:///./node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at p
    at Typography (webpack-internal:///./node_modules/@material-ui/core/esm/Typography/Typography.js:166:28)
    at WithStyles (webpack-internal:///./node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at DialogContentText
    at WithStyles (webpack-internal:///./node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at DialogContentText (webpack-internal:///./components/Dialog.tsx:164:19)
    at div
    at DialogContent (webpack-internal:///./node_modules/@material-ui/core/esm/DialogContent/DialogContent.js:43:23)
    at WithStyles (webpack-internal:///./node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at DialogContent (webpack-internal:///./components/Dialog.tsx:147:19)
    at div
    at Paper (webpack-internal:///./node_modules/@material-ui/core/esm/Paper/Paper.js:48:23)
    at WithStyles (webpack-internal:///./node_modules/@material-ui/styles/esm/withStyles/withStyles.js:61:31)
    at div
`

I think Typography must render a `<p>` under the hood and so must `DialogTextContent`.

I think you can just remove `<DialogTextContent>` here. That removes the error and doesn't change how it looks.

File: app/web/features/communities/events/CreateEventPage.test.tsx [hunk 2]
```diff
@@ -16,6 +16,23 @@ const createEventMock = service.events.createEvent as jest.MockedFunction<
...
+  timezone: "Australia/Broken_Hill",
+  hasStrongVerification: false,
+  birthdateVerificationStatus: 1,
+  genderVerificationStatus: 3,
+  doNotEmail: false,
+};
+
 describe("Create event page", () => {
   beforeAll(() => {
     server.listen();
```
Comment (nabramow) [inline app/web/features/communities/events/CreateEventPage.test.tsx:38]: How about adding a test to check that the popup comes up if there is no account profile?

You could mock the GetUser function to return what counts as an empty profile, then import the Dialog and check to see if it rendered.

File: app/web/features/dashboard/locales/en.json [hunk 3]
```diff
@@ -5,6 +5,18 @@
...
+    "actions": {
+      "send_request": "send a request",
+      "send_message": "send a message",
+      "create_event": "create an event"
+    },
+    "edit_profile_button": "Edit your profile now",
+    "cancel_button": "Never mind"
+  },
   "your_communities_heading": "Your communities",
   "all_communities_heading": "All communities",
```
Comment (nabramow) [inline app/web/features/dashboard/locales/en.json:21]: No Finnish?! ;-)

**Comment by nabramow** [comment]
This is working well for me! Just fix that console error and I'm happy to approve. Tests would be nice but I won't block you for it.

**Comment by aapeliv** [comment]
It now looks like this:

Screenshot from 2024-09-10 19-55-06

**Review by nabramow — DISMISSED** [review]
Awesome thanks for adding the test!

**Comment by aapeliv** [comment]
The right-margins weren't right, so I instead did two of these:

Screenshot from 2024-09-10 20-09-45

**Comment by nabramow** [comment]
> The right-margins weren't right, so I instead did two of these:
> 
> Screenshot from 2024-09-10 20-09-45

Oh yeah the buttons are not centered I think, now that I look at it.

**Pull request merged**